Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/service/ProductService.java

```diff
@@ -0,0 +1,67 @@
+package com.shopkeeper.productservice.service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.shopkeeper.productservice.dao.ProductServiceRepository;
+import com.shopkeeper.productservice.dto.Product;
+
+@Service
+public class ProductService {
+
+  @Autowired ProductServiceRepository productRepository;
+
+  public List<Product> getAllProducts() {
+    List<Product> products = new ArrayList<Product>();
+    productRepository.findAll().forEach(products::add);
+    return products;
+  }
+
+  public Product getProductById(Integer id) {
+    return productRepository.findById(id).orElse(null);
```

Comment:
this needs to be handled better mainly in terms of HttpCode. So, as suggested in the previous comment, if you move the logic of returning the ResponseEntity from here, you can accordingly select the status code as well.